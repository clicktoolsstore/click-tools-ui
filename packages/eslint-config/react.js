/** @type {import('eslint').Linter.Config} */
module.exports = {
  extends: ['@rocketseat/eslint-config/react'],
  rules: {
    'no-console': 'warn',
    'prettier/prettier': [
      'error',
      {
        printWidth: 80,
        tabWidth: 2,
        singleQuote: true,
        trailingComma: 'all',
        arrowParens: 'always',
        semi: false,
        endOfLine: 'auto',
        plugins: ['prettier-plugin-tailwindcss'],
        tailwindAttributes: ['class', 'className', 'ngClass', '.*ClassName'],
        tailwindFunctions: ['tv', 'twMerge', 'cn'],
      },
    ],
  },
}
