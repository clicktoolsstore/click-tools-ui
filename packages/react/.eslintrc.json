{
  "root": true,
  "extends": ["@click-tools-ui/eslint-config/react"],
  "overrides": [
    {
      // 3) Now we enable eslint-plugin-testing-library rules or preset only for matching testing files!
      "files": ["**/?(*.)+(spec|test).[jt]s?(x)"],
      "extends": [
        "plugin:testing-library/react",
        "plugin:jest-dom/recommended"
      ],
      "rules": { "@typescript-eslint/no-explicit-any": "off" }
    }
  ],
  "ignorePatterns": ["dist"],
  "rules": {
    "prettier/prettier": [
      "error",
      {
        "printWidth": 80,
        "tabWidth": 2,
        "singleQuote": true,
        "trailingComma": "all",
        "arrowParens": "always",
        "semi": false,
        "endOfLine": "auto",
        "plugins": ["prettier-plugin-tailwindcss"],
        "tailwindAttributes": ["class", "className", "ngClass", ".*ClassName"],
        "tailwindFunctions": ["tv", "twMerge", "cn"]
      }
    ]
  }
}
