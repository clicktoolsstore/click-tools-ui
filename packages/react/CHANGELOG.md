# @click-tools-ui/react

## 0.1.0

### Minor Changes

- base projects creation
