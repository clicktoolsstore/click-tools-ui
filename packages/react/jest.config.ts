import { Config } from 'jest'

const config: Config = {
  testEnvironment: 'jest-environment-jsdom',
  transform: {
    '^.+\\.(ts|tsx)$': 'ts-jest',
  },
  transformIgnorePatterns: ['/node_modules/'],
  moduleNameMapper: {
    '^~/(.*)$': '<rootDir>/src/$1',
  },
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
  coverageReporters: ['json-summary', 'text', 'lcov'],
  collectCoverageFrom: [
    '**/*.{ts,tsx}',
    '!**/src/index.tsx',
    '!**/jest.setup.ts',
    '!**/jest.config.ts',
  ],
}

export default config
