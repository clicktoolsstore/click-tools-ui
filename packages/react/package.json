{
  "name": "@click-tools-ui/react",
  "private": true,
  "version": "0.0.0",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "scripts": {
    "dev": "tsup src/index.tsx --format esm, cjs --dts --watch",
    "build": "tsup src/index.tsx --format esm, cjs --dts --external react",
    "lint": "eslint src/**/*.ts* --fix"
  },
  "dependencies": {
    "@stitches/react": "^1.2.8"
  },
  "devDependencies": {
    "@click-tools-ui/eslint-config": "*",
    "@click-tools-ui/tokens": "*",
    "@click-tools-ui/ts-config": "*",
    "@types/react": "^18.2.15",
    "@types/react-dom": "^18.2.7",
    "react": "^18.2.0",
    "tsup": "^7.2.0"
  }
}
