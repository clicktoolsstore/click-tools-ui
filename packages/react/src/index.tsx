export * from '~/components/Button'

export * from '~/components/Layouts'

export type { CSS } from '~/styles/index'
