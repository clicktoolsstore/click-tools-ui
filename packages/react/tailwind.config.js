import { colors } from '@click-tools-ui/tokens'

/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ['./src/**/*.{js,ts,jsx,tsx}'],
  theme: {
    colors,
  },
}
