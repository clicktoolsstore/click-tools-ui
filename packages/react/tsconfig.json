{
  "extends": "@click-tools-ui/ts-config/react.json",
  "compilerOptions": {
    "strict": false,
    "baseUrl": ".",
    "paths": {
      "~/*": ["./src/*"]
    }
  },
  "include": ["src"]
}
