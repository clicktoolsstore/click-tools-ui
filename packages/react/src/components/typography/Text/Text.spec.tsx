import { render, screen } from '@testing-library/react'
import { Text, TextProps, textStyle } from '.'

describe('Text Component', () => {
  it('should render correctly with default props', () => {
    const text = 'Some Text'
    render(<Text>{text}</Text>)

    const element = screen.getByText(text)
    expect(element).toBeInTheDocument()

    // check defaults
    expect(element.tagName).toBe('SPAN')

    expect(element).toHaveClass(
      'font-roboto text-neutral-900 text-base leading-1.3 font-normal',
      { exact: true },
    )
  })
  it('should render correctly "size" variants', () => {
    const sizes: Record<TextProps['size'], any> = {
      md: '',
      xs: '',
      '2xl': '',
    }
    const text = 'text-'

    render(
      <>
        {Object.keys(sizes).map((key) => (
          <Text key={key} size={key as TextProps['size']}>
            {text + key}
          </Text>
        ))}
      </>,
    )

    Object.keys(sizes).forEach((key) => {
      const element = screen.getByText(`${text}${key}`)

      expect(element).toHaveClass(textStyle.variants.size[key])
      expect(element.className).toMatchSnapshot(`size-${key}`)
    })
  })

  it('should render correctly "weight" variants', () => {
    const weights: Record<TextProps['weight'], any> = {
      medium: '',
      bold: '',
      regular: '',
    }
    const text = 'text-'

    render(
      <>
        {Object.keys(weights).map((key) => (
          <Text key={key} weight={key as TextProps['weight']}>
            {text + key}
          </Text>
        ))}
      </>,
    )

    Object.keys(weights).forEach((key) => {
      const element = screen.getByText(`${text}${key}`)

      expect(element).toHaveClass(textStyle.variants.weight[key])
      expect(element.className).toMatchSnapshot(`weight-${key}`)
    })
  })
  it('should render correctly "leading" variants', () => {
    const leading: Record<TextProps['leading'], any> = {
      '1.3': '',
      '1.6': '',
    }
    const text = 'text-'

    render(
      <>
        {Object.keys(leading).map((key) => (
          <Text key={key} leading={key as TextProps['leading']}>
            {text + key}
          </Text>
        ))}
      </>,
    )

    Object.keys(leading).forEach((key) => {
      const element = screen.getByText(`${text}${key}`)

      expect(element).toHaveClass(textStyle.variants.leading[key])
      expect(element.className).toMatchSnapshot(`leading-${key}`)
    })
  })

  it('should render the correct "as" tag element', () => {
    const as: Record<TextProps['as'], any> = {
      span: '',
      label: '',
      p: '',
      div: '',
    }
    const text = 'text-'

    render(
      <>
        {Object.keys(as).map((key) => (
          <Text key={key} as={key as TextProps['as']}>
            {text + key}
          </Text>
        ))}
      </>,
    )

    Object.keys(as).forEach((key) => {
      const element = screen.getByText(`${text}${key}`)

      expect(element.tagName).toBe(key.toUpperCase())
    })
  })
})
