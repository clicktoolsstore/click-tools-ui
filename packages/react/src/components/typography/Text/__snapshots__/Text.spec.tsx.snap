// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should render correctly "leading" variants: leading-1.3 1`] = `"font-roboto text-primary-900 text-base leading-1.3 font-normal"`;

exports[`Text Component should render correctly "leading" variants: leading-1.6 1`] = `"font-roboto text-primary-900 text-base leading-1.6 font-normal"`;

exports[`Text Component should render correctly "size" variants: size-2xl 1`] = `"font-roboto text-primary-900 text-2xl leading-1.3 font-normal"`;

exports[`Text Component should render correctly "size" variants: size-md 1`] = `"font-roboto text-primary-900 text-base leading-1.3 font-normal"`;

exports[`Text Component should render correctly "size" variants: size-xs 1`] = `"font-roboto text-primary-900 text-xs leading-1.3 font-normal"`;

exports[`Text Component should render correctly "weight" variants: weight-bold 1`] = `"font-roboto text-primary-900 text-base leading-1.3 font-bold"`;

exports[`Text Component should render correctly "weight" variants: weight-medium 1`] = `"font-roboto text-primary-900 text-base leading-1.3 font-medium"`;

exports[`Text Component should render correctly "weight" variants: weight-regular 1`] = `"font-roboto text-primary-900 text-base leading-1.3 font-normal"`;
